src/main/java/org/opensearch/geospatial/ip2geo/action/PutDatasourceAction.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.geospatial.ip2geo.action;
+
+import org.opensearch.action.ActionType;
+import org.opensearch.action.support.master.AcknowledgedResponse;
+
+/**
+ * GeoIp datasource creation action
+ */
+public class PutDatasourceAction extends ActionType<AcknowledgedResponse> {
+    /**
+     * Put datasource action instance
+     */
+    public static final PutDatasourceAction INSTANCE = new PutDatasourceAction();
```

Comment:
I know we have been using this way of creating the instance, but can we avoid this by may be using createComponents? or Guice framework for DI.

Reply:
We cannot without modifying the way OpenSearch core initialize the actions.